examples/twentyseven-pt-stencil-solver/twentyseven-pt-stencil-solver.cpp

```diff
@@ -0,0 +1,394 @@
+are permitted provided that the following conditions are met:
+
+1. Redistributions of source code must retain the above copyright notice,
+   this list of conditions and the following disclaimer.
+
+2. Redistributions in binary form must reproduce the above copyright notice,
+   this list of conditions and the following disclaimer in the documentation
+   and/or other materials provided with the distribution.
+
+3. Neither the name of the copyright holder nor the names of its contributors
+   may be used to endorse or promote products derived from this software
+   without specific prior written permission.
+
+THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS" AND
+ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE IMPLIED
+WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE
+DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE LIABLE FOR
+ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES
+(INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES;
+LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON
+ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+(INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS
+SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+******************************<GINKGO LICENSE>*******************************/
+
+/*****************************<COMPILATION>***********************************
+The easiest way to build the example solver is to use the script provided:
+./build.sh <PATH_TO_GINKGO_BUILD_DIR>
+
+Ginkgo should be compiled with `-DBUILD_REFERENCE=on` option.
+
+Alternatively, you can setup the configuration manually:
+
+Go to the <PATH_TO_GINKGO_BUILD_DIR> directory and copy the shared
+libraries located in the following subdirectories:
+
+    + core/
+    + core/device_hooks/
+    + reference/
+    + omp/
+    + cuda/
+
+to this directory.
+
+Then compile the file with the following command line:
+
+c++ -std=c++11 -o twentyseven-pt-stencil-solver
+twentyseven-pt-stencil-solver.cpp -I../.. \ -L. -lginkgo -lginkgo_reference
+-lginkgo_omp -lginkgo_cuda
+
+(if ginkgo was built in debug mode, append 'd' to every library name)
+
+Now you should be able to run the program using:
+
+env LD_LIBRARY_PATH=.:${LD_LIBRARY_PATH} ./twentyseven-pt-stencil-solver
+
+*****************************<COMPILATION>**********************************/
+
+#include <array>
+#include <chrono>
+#include <ginkgo/ginkgo.hpp>
+#include <iostream>
+#include <map>
+#include <string>
+#include <vector>
+
+/*
+double alpha_c = 38.0/6.0;
+double beta_c = -4.0/6.0;
+double gamma_c = -1.0/6.0;
+double delta_c = -1.0/24.0;
+*/
+const double alpha_c = 26;
+const double beta_c = -1;
+const double gamma_c = -1;
+const double delta_c = -1;
+
+std::array<double, 27> coefs{delta_c, gamma_c, delta_c, gamma_c, beta_c,
```

Comment:
Again, use `// clang-format on ` and `off` to format the number of rows and columns properly. 

If you could move this with to the main function and allow the user to select at run-time, that would be great.

Reply:
Yes. I thought about that, but it didn't seem important enough. I can change this.

I also can do this for the nine point example. Requires a minor rewrite for it.

Reply:
Done and pushed. nine point is still missing though